Patches/ChatCommandPatch.cs

```diff
@@ -1583,6 +1587,37 @@ static Color32 RndCLR()
                 default:
                     Main.isChatCommand = false;
                     break;
+
+                case "/spectate":
+                case "/спектейт":
+                case "/观战":
+                case "/espectar":
+                    if (!GameStates.IsLobby)
+                    {
+                        Utils.SendMessage(GetString("Message.OnlyCanUseInLobby"), PlayerControl.LocalPlayer.PlayerId);
+                        break;
+                    }
+

```

Comment:
Add options:
Max Spectator limit
Host Approval (this may be a little harder. go get a good ide)